xapian-applications/omega/omindexcheck.cc

```diff
@@ -92,8 +94,6 @@ index_test()
 		  {"Zannual", "Zfed", "Zreturn", "Zwhile"}});
 #endif
 #if defined HAVE_LIBARCHIVE
-    // blank file
-    tests.insert({"odf/blank.odt", {}});
```

Comment:
This removes a testcase, which seems undesirable.

Adding per-testcase flags would address this in a much more satisfactory way as it would allow marking the testcases where "skip for no terms" is appropriate.  Then we could keep this testcase, and we'd also catch "missing all terms" bugs for formats where this missing plugin situation doesn't apply (which is probably at least everything that doesn't use libextractor, and perhaps even everything except libextractor with particular plugins).

Reply:
Having a flag seems something really useful, but perhaps it would be better making that modification on a new PR (to avoid this become even larger).

Reply:
I have made the necessary changes here.
* Change implementation of testcase to a structure.
* Added an enum for flags with 3 values as of now - `FAIL_IF_NO_TERMS`, `SKIP_IF_NO_TERMS`, `PASS_IF_NO_TERMS`

Reply:
I don't think we need 3 flags for this.  If there are 3 flags, there are 8 possible combinations of them, but most of the combinations don't make any sense.  As a general design point, it's cleaner to have a design where there aren't invalid combinations like this to worry about.

I think there are really just 2 cases here:

* We're using an extractor plugin which may not extract anything due to a missing dependency - this is `SKIP_IF_NO_TERMS`
* We aren't - not specifying `SKIP_IF_NO_TERMS` covers this case (and if you add a constructor to `struct testcase` with the parameter for `flags` last and with a default of 0 then this you should be able to only specify the flag on the testcases that need it and avoid the clutter of flags on every testcase in the list).

Note that you can tell the cases that you currently have marked with `FAIL_IF_NO_TERMS` and `PASS_IF_NO_TERMS` apart because the latter has `terms.empty()` and the former don't.  It wouldn't really make sense to list terms but have `PASS_IF_NO_TERMS`, and a testcase listing no terms with `FAIL_IF_NO_TERMS` would have to always fail.

Reply:
I have changed this and have updated the PR.